src/content/reference/react-dom/components/common.md

```diff
@@ -282,6 +282,90 @@ To support backwards compatibility, if a cleanup function is not returned from t
 * When Strict Mode is on, React will **run one extra development-only setup+cleanup cycle** before the first real setup. This is a stress-test that ensures that your cleanup logic "mirrors" your setup logic and that it stops or undoes whatever the setup is doing. If this causes a problem, implement the cleanup function.
 * When you pass a *different* `ref` callback, React will call the *previous* callback's cleanup function if provided. If no cleanup function is defined, the `ref` callback will be called with `null` as the argument. The *next* function will be called with the DOM node.
 
+#### Troubleshooting: My ref points to a unmounted DOM node {/*my-ref-points-to-a-unmounted-dom-node*/}
```

Comment:
generally Troubleshooting is a top-level section that goes after Reference _and_ after Usage on every other page. i'm hesitant to add a one-off troubleshooting in the middle of a Reference. i'm also not convinced this specific problem is super common or super necessary to document in detail since it doesn't actually break things. 

i'd actually suggest killing this section and instead adding a Usage recipe to `useRef` page that describes how to _set a ref and also do something else_ — like Chance's original example of _what he was actually trying to do_. and show the correct example there.

Reply:
Sounds good to me. I didn't do this initially because the docs for a ref callback are here and not in useRef.

Do you think moving the "`ref` callback function" section from `/reference/react-dom/components/common.md` to  the useRef page make sense too? or just this "troubleshooting" content?

Reply:
Rewrote this PR to be a usage example in useRef for detecting DOM changes